pgjdbc/src/main/java/org/postgresql/jdbc/PgPreparedStatement.java

```diff
@@ -626,9 +627,15 @@ public void setObject(int parameterIndex, Object in, int targetSqlType, int scal
         setString(parameterIndex, castToString(in), Oid.BPCHAR);
         break;
       case Types.VARCHAR:
-      case Types.LONGVARCHAR:
         setString(parameterIndex, castToString(in), getStringType());
         break;
+      case Types.LONGVARCHAR:
+        if (in instanceof InputStream) {
+          preparedParameters.setText(parameterIndex, (InputStream)in);
```

Comment:
Thanks. Because the test database may not be UTF-8, I will restrict the test cases in the unit test to  printable ASCII. The test cases in ReaderInputStreamTest will cover the surrogate pairs.